tests/evaluate/test_evaluation.py

```diff
@@ -1316,3 +1316,28 @@ def test_evaluate_lightgbm_regressor():
     assert "mean_absolute_error" in run.data.metrics
     assert "mean_squared_error" in run.data.metrics
     assert "root_mean_squared_error" in run.data.metrics
+
+
+def test_evaluate_with_function_input_single_output():
```

Comment:
can we add a test with pandas dataframe too?

Reply:
X is already a pandas dataframe.
```python
In [8]: X
Out[8]:
          age       sex       bmi        bp        s1        s2        s3        s4        s5        s6
0    0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401 -0.002592  0.019907 -0.017646
5   -0.092695 -0.044642 -0.040696 -0.019442 -0.068991 -0.079288  0.041277 -0.076395 -0.041176 -0.096346
10  -0.096328 -0.044642 -0.083808  0.008101 -0.103389 -0.090561 -0.013948 -0.076395 -0.062917 -0.034215
15  -0.052738  0.050680 -0.018062  0.080401  0.089244  0.107662 -0.039719  0.108111  0.036060 -0.042499
20  -0.049105 -0.044642 -0.056863 -0.043542 -0.045599 -0.043276  0.000779 -0.039493 -0.011897  0.015491
..        ...       ...       ...       ...       ...       ...       ...       ...       ...       ...
420  0.023546 -0.044642 -0.036385  0.000079  0.001183  0.034698 -0.043401  0.034309 -0.033246  0.061054
425 -0.078165 -0.044642 -0.040696 -0.081413 -0.100638 -0.112795  0.022869 -0.076395 -0.020292 -0.050783
430 -0.009147 -0.044642 -0.056863 -0.050427  0.021822  0.045345 -0.028674  0.034309 -0.009919 -0.017646
435 -0.012780 -0.044642 -0.023451 -0.040099 -0.016704  0.004636 -0.017629 -0.002592 -0.038460 -0.038357
440 -0.045472 -0.044642  0.039062  0.001215  0.016318  0.015283 -0.028674  0.026560  0.044529 -0.025930

[89 rows x 10 columns]

In [9]: import pandas as pd

In [10]: type(X)
Out[10]: pandas.core.frame.DataFrame
```